src/custom/state/orders/utils.ts

```diff
@@ -268,3 +307,66 @@ export function getEstimatedExecutionPrice(
 
   return estimatedExecutionPrice
 }
+
+/**
+ * Gets the remainder amounts for both sell and buy, already discounting for surplus, if any surplus or matches
+ *
+ * Sell orders will have the surplus in the buy token.
+ * Which means sell amount does not need to be adjusted, while the buy amount does.
+ * Since the surplus is in the for of additional buy amount, we remove the surplus from the executed buy amount
+ *
+ * Buy orders will have the surplus in the sell token.
+ * Which means buy amount does not need to be adjusted, while the sell amount does.
+ * Since the surplus is in the for of less sell tokens being consumed, we add the surplus to the executed sell amount
+ *
+ * When there's no surplus it either means: there were no matches or it matched without any surplus
+ * In both cases, the sell and buy remainders can be returned in full
+ * @param order
+ */
+export function getRemainderAmountsWithoutSurplus(order: Order): { buyAmount: string; sellAmount: string } {
```

Comment:
Looks super clean and understandable!

Reply:
agree, thanks for such a nice explainer in the function on what it does